api/v1/api.go

```diff
@@ -18,48 +19,84 @@ func NewWrapper(vault vault.Storage) Wrapper {
 func (w Wrapper) DeleteSecret(ctx context.Context, request DeleteSecretRequestObject) (DeleteSecretResponseObject, error) {
 	err := w.vault.DeleteSecret(request.Key)
 	if err != nil {
+		if err == vault.ErrNotFound {
+			return DeleteSecret404JSONResponse{
+				Backend: backend,
+				Detail:  err.Error(),
+				Status:  404,
+				Title:   "could not delete secret",
```

Comment:
secret not found?

Reply:
The error detail contains the "key not found" text. I think the combined information is more clear.